v2/lints/mozilla/lint_mp_ecdsa_signature_encoding_correct.go

```diff
@@ -0,0 +1,127 @@
+
+When a root or intermediate certificate's ECDSA key is used to produce a signature, only the following algorithms may
+be used, and with the following encoding requirements:
+
+If the signing key is P-256, the signature MUST use ECDSA with SHA-256. The encoded AlgorithmIdentifier MUST match the
+following hex-encoded bytes: 300a06082a8648ce3d040302.
+
+If the signing key is P-384, the signature MUST use ECDSA with SHA-384. The encoded AlgorithmIdentifier MUST match the
+following hex-encoded bytes: 300a06082a8648ce3d040303.
+
+The above encodings consist of the corresponding OID with the parameters field omitted, as specified by RFC 5758,
+Section 3.2. Certificates MUST NOT include a NULL parameter. Note this differs from RSASSA-PKCS1-v1_5, which includes
+an explicit NULL.
+
+************************************************/
+
+import (
+	"bytes"
+	"encoding/asn1"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/util"
+)
+
+type ecdsaSignatureAidEncoding struct{}
+
+func (l *ecdsaSignatureAidEncoding) Initialize() error {
+	return nil
+}
+
+func (l *ecdsaSignatureAidEncoding) CheckApplies(c *x509.Certificate) bool {
+	// check for all ECDSA signatures to avoid missing this lint if an unsupported algorithm is used on the first place
+	// 1.2.840.10045.4.3.1 is SHA224withECDSA
+	return c.SignatureAlgorithm == x509.ECDSAWithSHA1 ||
+		c.SignatureAlgorithm == x509.ECDSAWithSHA256 ||
+		c.SignatureAlgorithm == x509.ECDSAWithSHA384 ||
+		c.SignatureAlgorithm == x509.ECDSAWithSHA512 ||
+		c.SignatureAlgorithmOID.Equal(asn1.ObjectIdentifier{1, 2, 840, 10045, 4, 3, 1})
+}
+
+func (l *ecdsaSignatureAidEncoding) Execute(c *x509.Certificate) *lint.LintResult {
+	// We must check consistency of the issuer public key to the signature algorithm
+	// (see for example: If the signing key is P-256, the signature MUST use ECDSA with SHA-256.
+	// The encoded AlgorithmIdentifier MUST match the following hex-encoded bytes: 300a06082a8648ce3d040302.)
+	// Thus we need the issuer public key which it is not available so easy.
+	// At this stage all certificates (also of sub-CAs and root-CAs, provided they are linted) are either
+	// P-256 or P-384 (see lint e_mp_ecdsa_pub_key_encoding_correct).
+	// Therefore we check the length of the signature in the certificate. If it is less that 72 bytes then it is
+	// done by a P-256 key and if it is less than 104 bytes it is done by a P-384.
+
+	signature := c.Signature
+	signatureSize := len(signature)
+	encoded, err := util.GetSignatureAlgorithmInTBSEncoded(c)
+
+	if err != nil {
+		return &lint.LintResult{Status: lint.Error, Details: "error reading signatureAlgorithm from TBS"}
+	}
+
+	// Signatures made with P-256 are not going to be greater that 72 bytes long
+	// Seq Tag+Length = 2, r Tag+length = 2, s Tag+length =2, r max 32+1 (unsigned representation), same for s
```

Comment:
nit: consistent `=` spacing.
```suggestion
	// Seq Tag+Length = 2, r Tag+length = 2, s Tag+length = 2, r max 32+1 (unsigned representation), same for s
```

Reply:
Done.